### DISCUSSION TITLE
sd_output() add type="label_question" and "label_value"

### DISCUSSION THREAD

**Discussion by StefanMunnes** [surveydown-dev/surveydown#128] [Feature Requests]
### Description

I think it would be also useful to be able to output the question label and probably more often the label of the selected value via the `sd_output()` function. Feel free to change the type wording. Example:

SURVEY.QMD
```md
```{r}
sq_question(
  type = "mc",
  id = "fruits",
  label = "What is your favorite fruit?",
  option = c(
    "Apple" = "apple",
    "Banana" = "banana",
    "Dragon Fruit" = "dragon_fruit",
    "Blood Orange" = "blood_orange"
  )
{```}
You answered for the question: "`sd_output(id = "fruits", type = "label_question")`" that you like "`sd_output(id = "fruits", type = label_value")`s" the most.
```

**Comment by jhelvy** [comment]
Good idea. Just to clarify, the markdown you wrote above should render something this if the user chose the `"apple"` option:

> You answered for the question: "What is your favorite fruit?" that you like "Apple `s" the most.

So you're asking to not just be able to insert the chosen _value_ (e.g. `apple`) but also the question label as well as the option label. This is a good idea, especially for the option label as it's more likely than not that you would want to display the label rather than the value for the option. 

As for the argument names, I like `"label_question"` and `"label_option"` as these are the names we use to define the elements in `sd_question()`. `"label_value"` seemed a little confusing to me as we already have `"value"` as one of the `type` options. 

We can put this on our todo list. Of course, if you want to try and do it yourself, there are two things you need to change:

1. You would need to extract each label (the question and option labels) and then make them available in the output. You can see how we do this with the `"value"` type [here](https://github.com/surveydown-dev/surveydown/blob/main/R/server.R#L277C1-L278C84). The key line there is this: `output[[paste0(local_id, "_value")]] <- renderText({ formatted_value })`. What this is doing is storing the rendered value in the output. We do this so the user doesn't have to - it just automatically gets put into the output. So you would need to do the same for the question label and the option label. I'm not yet sure exactly how to extract those pieces.
2. You would need to add these options for the `type` argument to `sd_output()`. The code [here](https://github.com/surveydown-dev/surveydown/blob/main/R/ui.R#L823C1-L837C10) shows how we handle it for `type = "value"`. The key piece is `value_id <- paste0(id, "_value")`. Here we're automatically appending `_value` to the `id` to make sure that we return the value, not the question itself (which is what the `id` refers to).  So you'd need to add additional `if` conditions for `type = "label_question"` and `type = "label_option"` and return a similar output.

**Reply by StefanMunnes** [reply]
Regarding your question for clarification, yes, that is the example output I was thinking of. I also added the "dragon fruit" as an example with a space, which I assumed would be impossible to have as a value.

If it hasn't been done after the weekend, I'd like to see how to implement this feature. I already took a quick look in the code.
Should I clone the repository and make a pull request?

**Reply by jhelvy** [reply]
Yes if you want to give it a try, I would fork the repo, make your edits, then send a PR. Happy to review it! I think this should be rather straightforward. The only thing I'm not sure how to solve is how to actually extract the labels from the questions. The bit in the `sd_server` code that renders the _value_ (`renderText({ formatted_value })`) is easy because shiny inputs are designed to return the chosen input. They're not really designed to hand you the chosen _value label_, nor the question label. So solving that might require some clever thinking.

**Reply by StefanMunnes** [reply]
It was tough, but I opened a [pull request](https://github.com/surveydown-dev/surveydown/pull/131). Hope you like this solution.

**Reply by jhelvy** [reply]
Just saw, this is awesome! I'll try to look through it soon. Hooray for open source development!

**Comment by jhelvy** [comment]
This is now in the main branch from PR #131, with the version bumped to v0.3.7. Thanks again for adding this!
